### PR TITLE
[CINFRA-423] Replicated State Metrics Object

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose
This adds metrics for replicated states. It creates the infrastructure needed to add more metrics later on.

### CODE REVIEW

File: Documentation/Metrics/arangodb_replication2_replicated_state_leader_number.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+help: |
+  Number of leader replicated states.
+unit: number
+type: counter
+category: Replication
+complexity: simple
+exposedBy:
+  - dbserver
+description: |
+  Number of replicated states this server is a leader for.
```
Comment (ansoboleva) [inline Documentation/Metrics/arangodb_replication2_replicated_state_leader_number.yaml:12]: Number of replicated states this server is a leader of.

File: Documentation/Metrics/arangodb_replication2_replicated_state_follower_number.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+help: |
+  Number of follower replicated states.
+unit: number
+type: counter
+category: Replication
+complexity: simple
+exposedBy:
+  - dbserver
+description: |
+  Number of replicated states this server is a follower for.
```
Comment (ansoboleva) [inline Documentation/Metrics/arangodb_replication2_replicated_state_follower_number.yaml:12]: Number of replicated states this server is a follower of.

File: Documentation/Metrics/arangodb_replication2_replicated_state_follower_number.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+help: |
+  Number of follower replicated states.
+unit: number
+type: counter
+category: Replication
+complexity: simple
+exposedBy:
+  - dbserver
+description: |
+  Number of replicated states this server is a leader of.
```
Comment (ansoboleva) [inline Documentation/Metrics/arangodb_replication2_replicated_state_follower_number.yaml:12]: Looks like you switched leader and follower in the updated descriptions :)

Comment (maierlars) [reply Documentation/Metrics/arangodb_replication2_replicated_state_follower_number.yaml]: Thanks for spotting!

**Review by ansoboleva — APPROVED** [review]
LGTM!

**Review by markuspf — APPROVED** [review]
LGTM

**Pull request merged**